src/main/scala/com/example/solution/dto/reservation.scala

```diff
@@ -0,0 +1,24 @@
+package com.example.solution.dto
+
+import com.example.solution.dto.screening.ScreeningData
+
+import java.time.LocalDateTime
+
+object reservation {
+
+  case class PickedSeat(row : Int, seatInRow : Int, ticketType : String)
+
+  case class ReservationData(
+                            screeningId : Int,
+                            pickedSeats: List[PickedSeat],
```

Comment:
are we able to create ReserwationData with empty list of picked seats? Do you know a solution which will help us to achieve the empty list validation at type level?

Reply:
We could use NonEmptyList datatype, which takes at least one element in its constructor datatype.